Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -324,10 +320,15 @@ class CombinatorialKalmanFilter {
     using BoundState = std::tuple<parameters_t, BoundMatrix, double>;
     using CurvilinearState =
         std::tuple<CurvilinearTrackParameters, BoundMatrix, double>;
-    // The source link container type
     /// Broadcast the result_type
     using result_type = CombinatorialKalmanFilterResult<traj_t>;
 
+    /// Fitted states that the actor has handled.
+    traj_t* fittedStates{nullptr};
+
+    /// This is used internally to store candidate trackstates
+    std::shared_ptr<traj_t> stateBuffer;
+
```

Comment:
right I changed this very early on while refactoring and this ended up not being necessary after all

the "result" (actor state) lives now within the propagator state and I had troubles to set state variables from the outside which is necessary in this case. sometimes we do have state in the actor but I guess just immutable? might be good to get rid of these all the way and to put it into the "result".

I surrounded this issue now by having a `makeState` method in the `Propagator` which lets you create a state, then mutate it and then call the propagator on this state. I think that covers our use cases.

having a `propagate` method which can be called multiple times also lets us steer the `Propagator` from the outside which will come handy for track finding in the future

Reply:
I am still not too happy with the name "result" but I kept it for now as it is still returned after the propagation for the usual convenience functions

Reply:
So the thinking was if it's immutable it can be a member, if it's mutable it's *state*. We didn't originally foresee actor-specific state, but started abusing the result for this early on.

My thinking was that we should eventually (tm) just rename *result* into *state* or *actorState* or whatever. This is orthogonal to putting said state into the main propagator state or not.